atlasdb-client/src/main/java/com/palantir/atlasdb/table/description/render/StreamStoreRenderer.java

```diff
@@ -577,15 +581,27 @@ private void storeBlocksAndGetFinalMetadata() {
             private void loadStreamWithCompression() {
                 line("@Override");
                 line("public InputStream loadStream(Transaction t, final ", StreamId, " id) {"); {
-                    line("return new LZ4BlockInputStream(super.loadStream(t, id));");
+                    line("try {"); {
+                        line("return new " + clientSideCompression.inputClass + "(super.loadStream(t, id));");
+                    } line("} catch (IOException e) {"); {
```

Comment:
If an exception is thrown here, this will leak the stream returned by super.loadStream